src/nvim/lua/executor.c

```diff
@@ -219,8 +222,11 @@ static int nlua_luv_cfpcall(lua_State *lstate, int nargs, int nresult, int flags
     }
     const char *error = lua_tostring(lstate, -1);
 
-    multiqueue_put(main_loop.events, nlua_luv_error_event,
-                   xstrdup(error), (void *)(intptr_t)kCallback);
+    if (!event_error_queued) {
+      multiqueue_put(main_loop.events, nlua_luv_error_event,
+                     xstrdup(error), (void *)(intptr_t)kCallback);
+      event_error_queued = true;
```

Comment:
Coalescing multiple error callbacks into one might be good but then we need to preserve the individual `error` messages and display all of them at  once we are ready to handle the callback.

Reply:
This is to avoid recursiveness. If one msg_show event handler itself _causes_ an error we need to avoid emitting that error multiple times, which presumably will be the same every time.

But yeah I suppose this solution might hide multiple different errors.

Reply:
This and possibly other recursion checks are no longer necessary because I added automatic detachment for when a vim.ui_attach callback produces excessive errors.